### PR TITLE
improve bad-regions and tombstone subcommands for tikv-ctl

### PR DESCRIPTION
**Pull request by hicqu**
## What have you changed? (mandatory)
- [x] bad-regions covers more cases now
- [x] tombstone has a new option "force" to skip configuration changes

### CODE REVIEW

File: src/server/debug.rs [hunk 1]
```diff
@@ -344,6 +344,28 @@ impl Debugger {
         Ok(errors)
     }
 
+    pub fn set_region_tombstone_by_id(&self, regions: Vec<u64>) {
```
Comment (zhangjinpeng1987) [inline src/server/debug.rs]: Please add test for this function.

File: src/server/debug.rs [hunk 2]
```diff
@@ -344,6 +344,28 @@ impl Debugger {
...
+        let db = &self.engines.kv;
+        let wb = WriteBatch::new();
+        for region_id in regions {
+            let key = keys::region_state_key(region_id);
+            let region_state = db
+                .get_msg_cf::<RegionLocalState>(CF_RAFT, &key)
+                .unwrap()
+                .unwrap();
+            if region_state.get_state() == PeerState::Tombstone {
+                println!("skip because it's already tombstone");
```
Comment (overvenus) [inline src/server/debug.rs]: Please use `v1`.

File: src/server/debug.rs [hunk 3]
```diff
@@ -462,6 +484,9 @@ impl Debugger {
             if raft_state.get_last_index() < apply_state.get_applied_index() {
                 return Err(Error::Other("last index < applied index".into()));
             }
+            if raft_state.get_hard_state().get_commit() < apply_state.get_applied_index() {
```
Comment (overvenus) [inline src/server/debug.rs]: Do we need to check `commit_index > last_index`?

File: src/server/debug.rs [hunk 4]
```diff
@@ -344,6 +344,28 @@ impl Debugger {
...
+            let region_state = db
+                .get_msg_cf::<RegionLocalState>(CF_RAFT, &key)
+                .unwrap()
+                .unwrap();
+            if region_state.get_state() == PeerState::Tombstone {
+                println!("skip because it's already tombstone");
+                return;
+            }
+            let region = &region_state.get_region();
+            write_peer_state(db, &wb, region, PeerState::Tombstone, None).unwrap();
```
Comment (overvenus) [inline src/server/debug.rs]: Do we need to write `MergeState`?

File: src/server/debug.rs (hunk 3 above)
Comment (hicqu) [reply src/server/debug.rs]: It's not necessary because they are in 1 rocksdb.

File: src/server/debug.rs (hunk 4 above)
Comment (hicqu) [reply src/server/debug.rs]: It's not necessary because leave it in merging status is fine.

File: src/server/debug.rs (hunk 3 above)
Comment (overvenus) [reply src/server/debug.rs]: If this check fails, it means either rocksdb is corrupted or there is a bug in raft.

File: src/server/debug.rs [hunk 5]
```diff
@@ -1809,6 +1845,28 @@ mod tests {
...
+
+        // tombstone region 1 which currently not exists.
+        let errors = debugger.set_region_tombstone_by_id(vec![1]).unwrap();
+        assert!(!errors.is_empty());
+
+        // region 1 with peers at stores 11, 12, 13.
+        init_region_state(engine, 1, &[11, 12, 13]);
+
+        // tombstone region 1.
+        let errors = debugger.set_region_tombstone_by_id(vec![1]).unwrap();
```
Comment (zhangjinpeng1987) [inline src/server/debug.rs]: Should get region_state after tombstone.

File: src/bin/tikv-ctl.rs [hunk 6]
```diff
@@ -1332,14 +1356,19 @@ fn main() {
...
                         .short("p")
                         .takes_value(true)
                         .multiple(true)
                         .use_delimiter(true)
                         .require_delimiter(true)
                         .value_delimiter(",")
                         .help("PD endpoints"),
+                )
+                .arg(
+                    Arg::with_name("force")
```
Comment (zhangjinpeng1987) [inline src/bin/tikv-ctl.rs]: I think local-mode is better.

**Comment by zhangjinpeng1987** [comment]
rest LGTM

**Pull request merged**